lib/autocomplete-manager.coffee

```diff
@@ -110,26 +113,59 @@ class AutocompleteManager
     return if @isCurrentFileBlackListed()
     cursor = @editor.getLastCursor()
     return unless cursor?
-    cursorPosition = cursor.getBufferPosition()
-    currentScope = cursor.getScopeDescriptor()
-    return unless currentScope?
-    currentScopeChain = currentScope.getScopeChain()
-    return unless currentScopeChain?
-
-    options =
-      editor: @editor
-      buffer: @buffer
-      cursor: cursor
```

Comment:
For now, they can do something simple like this. No need to reimplement that method

```coffee
regex = /[\w]+$/
line = editor.getTextInRange([[position.row, 0], position])
line.match(regex)?[0]
```

How should we provide that method to the provider? I dont think we should bake a prefix method into the cursor. The method being used has a different purpose. If they really really need the cursor, they can get it via `editor.getLastCursor()`. 

The parameters being passed provide parity with the old API. We can add more things later.